app/api/margatsni/v1/users/users.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Margatsni
+  module V1
+    module Users
+      class Users < Margatsni::V1::BaseV1
+        helpers do
+          def represent_user_with_token(user)
+            present user, with: Margatsni::V1::Entities::User
+          end
+        end
+
+        namespace :users do
+          desc 'register a user'
+          params do
+            requires :username, type: String
+            requires :email, type: String, regexp: /\A[^@\s]+@[^@\s]+\z/
```

Comment:
You have used regexp `/\A[^@\s]+@[^@\s]+\z/` in at least 3 places. If you would need to modify it, you will also need to do this everywhere. Let's create a const variable in the User class to avoid this.
`EMAIL_REGEXP=/\A[^@\s]+@[^@\s]+\z/`
Then you can use it like:
`requires :email, type: String, regexp: User::EMAIL_REGEXP`